cmd/minikube/cmd/start.go

```diff
@@ -1244,26 +1245,22 @@ func validateFlags(cmd *cobra.Command, drvName string) {
 	validateInsecureRegistry()
 }
 
-// This function validates that the --ports are not below 1024 for the host and not outside range
+// validatePorts validates that the --ports are not below 1024 for the host and not outside range
 func validatePorts(ports []string) error {
-	for _, portDuplet := range ports {
-		parts := strings.Split(portDuplet, ":")
-		if len(parts) > 2 {
-			ip := parts[0]
-			if net.ParseIP(ip) == nil {
-				return errors.Errorf("Sorry, the IP address provided with --ports flag is invalid: %s", ip)
-			}
-			parts = parts[1:]
-		}
-		for i, port := range parts {
-			p, err := strconv.Atoi(port)
+	_, portBindingsMap, err := nat.ParsePortSpecs(ports)
```

Comment:
Hi @tyabu12, can you confirm that this third-party library function offers improved functionality over the original code?

Reply:
@klaases 
docker-cli use the third-party library function to parse publish ports option.
https://github.com/docker/cli/blob/master/cli/command/container/opts.go#L416

Strictly speaking, convertToStandardNotation function converts target and published format at L411 (e.g.  `target=10,published=20` -> `20:10/tcp`), but the format seems to be unfamiliar.